src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -2295,15 +2295,33 @@ mobile_apis::Result::eType MessageHelper::VerifyImage(
     full_file_path += file_name;
   }
 
+  image[strings::value] = full_file_path;
   if (!file_system::FileExists(full_file_path)) {
     return mobile_apis::Result::INVALID_DATA;
   }
 
-  image[strings::value] = full_file_path;
-
   return mobile_apis::Result::SUCCESS;
 }
 
+mobile_apis::Result::eType MessageHelper::VerifyImage(
+    smart_objects::SmartObject& image,
+    ApplicationConstSharedPtr app,
+    ApplicationManager& app_mngr) {
+  smart_objects::SmartObject img = image;
+  const uint32_t image_type = image[strings::image_type].asUInt();
+  mobile_apis::ImageType::eType type =
+      static_cast<mobile_apis::ImageType::eType>(image_type);
+  mobile_apis::Result::eType result;
+
+  result = VerifyImageApplyPath(img, app, app_mngr);
```

Comment:
@tkireva It seems that `result` is not changed after the initialization, so it can be made `const`. The same applies to `type` variable above, where a name like `image_type` will be more informative.